swap_meet/vendor.py

```diff
@@ -1,2 +1,117 @@
+            self.swap_items(other_vendor, user_first_item, friend_first_item)
+            # self.inventory.remove(user_first_item)
+            # other_vendor.inventory.remove(friend_first_item)
+            # other_vendor.inventory.append(user_first_item)
+            # self.inventory.append(friend_first_item)
+            return True
+        
+    def get_by_category(self, category):
+        # Note: category of type string
+        # Create an empty list called "resulting_list"
+        resulting_list = []
+        # Loop through each Item instance in inventory
+        for item in self.inventory:
+            # If an instance's category (aka class name) same as provided category,
+            # add this instance to resulting_list
+            if item.get_category() == category:
+                resulting_list.append(item)
+        # Return updated resulting_list
+        return resulting_list
+    
+    def get_best_by_category(self, category):
+        # Call on get_by_category method to get a list of instances matching provided category
+        resulting_list = self.get_by_category(category)
+        # If resulting_list is empty, aka no matching instances, return None
+        if len(resulting_list) == 0:
+            return None
+        else:
+            # Otherwise, grab first instance in resulting_list and store in best_item
+            best_item = resulting_list[0]
+            # Read first instance's attribute condition and store in best_condition
+            best_condition = resulting_list[0].condition
+            # Loop thru each instance in resulting_list
+            for item in resulting_list:
+                # If an instance's attribute condition shares same value as best_condition's value
+                # re-assign this instance to best_item and its attribute condition to best_condition
+                if item.condition > best_condition:
+                    best_item = item
+                    best_condition = item.condition
+            # Return updated instance with best condition matching provided category
+            return best_item    
+        
+    def swap_best_by_category(self, other_vendor, my_priority, their_priority):
+        # Call on get_best_by_category method to get user's best instance matching friend's category
+        # and to get friend's best instance matching user's category
+        user_best_match = self.get_best_by_category(their_priority)
+        other_best_match = other_vendor.get_best_by_category(my_priority)
+        # If user has no instance matching friend's category or vice versa, return False
+        if user_best_match is None or other_best_match is None:
+            return False
+        else: 
+            # Otherwise, call on swap_items method to swap user's and friend's 
+            # matching best instances with each other 
+            self.swap_items(other_vendor, user_best_match, other_best_match)
+            # return True
+            return True
```

Comment:
Since `swap_items` can handle items that are `None`, you could also defer to it:

```python
        return self.swap_items(other_vendor, user_best_match, other_best_match)
```